File: spoj/GAMES.py
import math
from decimal import Decimal

def gcd(a, b):
    if b > a:
        return gcd(b, a)

    if a % b == 0:
        return b

    return gcd(b, a % b)

T = int(raw_input())

for i in range (0, T):
	number = Decimal(raw_input())
	number_dec_str = str(number-int(number))[1:]

	length = len(number_dec_str)
	ten = Decimal(math.pow(10, length-1))
	number = number*ten

	gcd_ = gcd(number, ten)

	print int(ten/gcd_)

File: spoj/GAMES.cpp
#include <stdio.h>
#include <math.h>
#include <string>
#include <iostream>

using namespace std;

long gcd(long a,long b)
{
	return b == 0 ? a : gcd(b , a % b);
}

int main(int argc, char const *argv[])
{
	int T;
	scanf("%d", &T);

	while (T-- > 0)
	{
		string s;

		cin >> s;
		
		int precision = 0;
		bool calculatePrecision = false;
		long number = 1;

		for(int i = 0; i < s.length(); i++)
		{
			char c = s[i];
			if (calculatePrecision)
			{
				precision += 1;
				number = number*10 + (c-48);
				continue;
			}

    		if (c == '.')
    		{
    			calculatePrecision = true;
    			continue;
    		}

    		number = number*10 + (c-48);
		}

		long ten = pow(10, precision);
		long gcd_ = gcd(number, ten);

		long ans = (long)ten/gcd_;

		printf("%ld\n", ans);

	}
	return 0;
}